File: 3. Sophomore_HKI/OOP/OOP_CUOIKI/QuanLyShopQuanAo (C#)/src/mypack/HoaDonNhapHang.java
package mypack;

public class HoaDonNhapHang {
	
	public String maHoaDonNhapHang;		// primary key 
	
	// foreign key
    public String maNoiCungCap;
    public String maNhanVien;

    // singular
    public NoiCungCap noiCungCap;
    public NhanVien nhanVien;

    // constructor
    public HoaDonNhapHang() { }
    public HoaDonNhapHang(HoaDonNhapHang hd)
    {
        this.maHoaDonNhapHang = hd.maHoaDonNhapHang;
        this.maNoiCungCap = hd.maNoiCungCap;
        this.maNhanVien = hd.maNhanVien;
        this.noiCungCap = hd.noiCungCap;
        this.nhanVien = hd.nhanVien;
    }
    public HoaDonNhapHang(String maHoaDonNhapHang, String maNoiCungCap, String maNhanVien,
        NoiCungCap noiCungCap, NhanVien nhanVien)
    {
        this.maHoaDonNhapHang = maHoaDonNhapHang;
        this.maNoiCungCap = maNoiCungCap;
        this.maNhanVien = maNhanVien;
        this.noiCungCap = noiCungCap;
        this.nhanVien = nhanVien;
    }
    
    // getter - setter
    public String getMaHoaDonNhapHang() {
		return maHoaDonNhapHang;
	}
	public void setMaHoaDonNhapHang(String maHoaDonNhapHang) {
		this.maHoaDonNhapHang = maHoaDonNhapHang;
	}
	public String getMaNoiCungCap() {
		return maNoiCungCap;
	}
	public void setMaNoiCungCap(String maNoiCungCap) {
		this.maNoiCungCap = maNoiCungCap;
	}
	public String getMaNhanVien() {
		return maNhanVien;
	}
	public void setMaNhanVien(String maNhanVien) {
		this.maNhanVien = maNhanVien;
	}
	public NoiCungCap getNoiCungCap() {
		return noiCungCap;
	}
	public void setNoiCungCap(NoiCungCap noiCungCap) {
		this.noiCungCap = noiCungCap;
	}
	public NhanVien getNhanVien() {
		return nhanVien;
	}
	public void setNhanVien(NhanVien nhanVien) {
		this.nhanVien = nhanVien;
	}   
}

File: 3. Sophomore_HKI/OOP/OOP_CUOIKI/QuanLyShopQuanAo (C#)/HoaDonNhapHang.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuanLyShopQuanAo
{
    public class HoaDonNhapHang
    {
        public string maHoaDonNhapHang { get; set; }     // primary key
        public DateTime ngayNhap { get; set; }
        public double donGia { get; set; }

        // foreign key
        public string maNoiCungCap { get; set; }
        public string maNhanVien { get; set; }

        // singular
        public NoiCungCap noiCungCap { get; set; }
        public NhanVien nhanVien { get; set; }

        // plural
        public ICollection<ChiTietHoaDonNhapHang> dsChiTiet_HoaDonNhap { get; set; }

        // constructor
        public HoaDonNhapHang() { }
        public HoaDonNhapHang(HoaDonNhapHang hoaDonNhap)
        {
            this.maHoaDonNhapHang = hoaDonNhap.maHoaDonNhapHang;
            this.ngayNhap = hoaDonNhap.ngayNhap;
            this.donGia = hoaDonNhap.donGia;
            this.maNoiCungCap = hoaDonNhap.maNoiCungCap;
            this.maNhanVien = hoaDonNhap.maNhanVien;
            this.noiCungCap = hoaDonNhap.noiCungCap;
            this.nhanVien = hoaDonNhap.nhanVien;
            this.dsChiTiet_HoaDonNhap = hoaDonNhap.dsChiTiet_HoaDonNhap;
        }
        public HoaDonNhapHang(string maHoaDonNhapHang, DateTime ngayNhap, double donGia, 
            string maNoiCungCap, string maNhanVien,NoiCungCap noiCungCap, 
            NhanVien nhanVien, ICollection<ChiTietHoaDonNhapHang> dsChiTiet_HoaDonNhap)
        {
            this.maHoaDonNhapHang = maHoaDonNhapHang;
            this.ngayNhap = ngayNhap;
            this.donGia = donGia;
            this.maNoiCungCap = maNoiCungCap;
            this.maNhanVien = maNhanVien;
            this.noiCungCap = noiCungCap;
            this.nhanVien = nhanVien;
            this.dsChiTiet_HoaDonNhap = dsChiTiet_HoaDonNhap;
        }
    }
}
